### PR TITLE
Remember tab + updates

### PR DESCRIPTION
**Pull request by rsnyman**
- On the Run and Result pages, "remember" the selected tab via the URL
- Update PatternFly React
- Make tabs "boxed"
- Fix up some style issues

### CODE REVIEW

File: frontend/src/components/result.js [hunk 1]
```diff
@@ -175,9 +215,9 @@ export class ResultView extends React.Component {
     return (
       <React.Fragment>
         {this.state.testResult &&
-        <Tabs activeKey={this.state.artifactsTabKey} onSelect={this.handleArtifactsTabClick}>
+        <Tabs activeKey={activeTab} onSelect={this.onTabSelect} isBox>
```
Comment (john-dupuy) [inline frontend/src/components/result.js:218]: Can we also add `isBox` to the `Tabs` on the Jenkins Job analysis view?
https://github.com/ibutsu/ibutsu-server/blob/master/frontend/src/views/jenkinsjobanalysis.js#L210

**Review by john-dupuy — APPROVED** [review]
Nice enhancement! Just one small comment.

File: frontend/src/components/result.js (hunk 1 above)
Comment (rsnyman) [reply frontend/src/components/result.js:218]: Do you want me to implement the tab remembering on this page too?

Comment (john-dupuy) [reply frontend/src/components/result.js:218]: If you'd like to. We can always do it as a follow-up.

Comment (rsnyman) [reply frontend/src/components/result.js:218]: Might as well :-)

Comment (rsnyman) [reply frontend/src/components/result.js:218]: Done :-)

**Pull request merged**